avocado_vt/test.py

```diff
@@ -213,21 +213,25 @@ def verify_background_errors(self):
         Logs all errors in the background_errors into background-error.log and
         error the test.
         """
-        err_file_path = os.path.join(self.logdir, BG_ERR_FILE)
         bg_errors = self.background_errors.get_all()
-        error_messages = ["BACKGROUND ERROR LIST:"]
-        for index, error in enumerate(bg_errors):
-            error_messages.append(
-                "- ERROR #%d -\n%s" % (index, "".join(
-                    traceback.format_exception(*error)
-                    )))
-        genio.write_file(err_file_path, '\n'.join(error_messages))
         if bg_errors:
-            msg = ["Background error"]
-            msg.append("s are" if len(bg_errors) > 1 else " is")
-            msg.append((" detected, please refer to file: "
-                        "'%s' for more details.") % BG_ERR_FILE)
-            self.error(''.join(msg))
+            err_file_path = os.path.join(self.logdir, BG_ERR_FILE)
+            try:
+                error_messages = ["BACKGROUND ERROR LIST:"]
+                for index, error in enumerate(bg_errors):
+                    error_messages.append(
+                        "- ERROR #%d -\n%s" % (index, "".join(
+                            traceback.format_exception(*error)
+                            )))
+                genio.write_file(err_file_path, '\n'.join(error_messages))
```

Comment:
Big try, it looks only need to try here.

Reply:
Beside this one, we should put `traceback.format_exception(*error)` into the `try` clause as well.

Reply:
Thanks for your review,  @luckyh  @xutian  How do you think about :
```
217         if bg_errors:                                                                                                 
218             err_file_path = os.path.join(self.logdir, BG_ERR_FILE)              
219             error_messages = ["BACKGROUND ERROR LIST:"]                         
220             try:                                                                
221                 for index, error in enumerate(bg_errors):                       
222                     error_messages.append(                                      
223                         "- ERROR #%d -\n%s" % (index, "".join(                  
224                             traceback.format_exception(*error)                  
225                             )))                                                 
226                 genio.write_file(err_file_path, '\n'.join(error_messages))      
227             except:                                                             
228                 self.log.warn("Failed to save background error: %s"             
229                               % error_messages)                                 
230             finally:                                                            
231                 msg = ["Background error"]                                      
232                 msg.append("s are" if len(bg_errors) > 1 else " is")            
233                 msg.append((" detected, please refer to file: "                 
234                             "'%s' for more details.") % BG_ERR_FILE)            
235                 self.error(''.join(msg)) 
```

Reply:
Not a good solution.  In case, save error info to file failed but promote "please refer to file" at last.